src/Engine/ProtoCore/Namespace/ElementRewriter.cs

```diff
@@ -11,13 +12,15 @@
 namespace ProtoCore.Namespace
 {
 
-    public class ElementRewriter
+    public class ElementRewriter : AstReplacer
     {
         private readonly ClassTable classTable;
-        
-        internal ElementRewriter(ClassTable classTable)
+        private readonly ElementResolver elementResolver;
+
+        internal ElementRewriter(ClassTable classTable, ElementResolver resolver)
         {
             this.classTable = classTable;
+            this.elementResolver = resolver;
```

Comment:
The `elementResolver` stores a map of a short name vs. the fully resolved name for a class. So that if the user imports a new library into Dynamo that has a class conflict with the first, the `elementResolver` can be queried to return its fully resolved name. This is to prevent older nodes from suddenly breaking upon importing new libraries. This cannot be done by the class table lookup as it will return a namespace conflict. The info therefore needs to be cached.